src/generators/stats.js

```diff
@@ -0,0 +1,75 @@
+import { renderStatsCard } from "../cards/stats-card.js";
+import { blacklist } from "../common/blacklist.js";
+import { parseArray, parseBoolean, renderError } from "../common/utils.js";
+import { fetchStats } from "../fetchers/stats-fetcher.js";
+import { isLocaleAvailable } from "../translations.js";
+
+export default async (options) => {
+  const {
+    username,
+    hide,
+    hide_title,
+    hide_border,
+    card_width,
+    hide_rank,
+    show_icons,
+    count_private,
+    include_all_commits,
+    line_height,
+    title_color,
+    ring_color,
+    icon_color,
+    text_color,
+    text_bold,
+    bg_color,
+    theme,
+    exclude_repo,
+    custom_title,
+    locale,
+    disable_animations,
+    border_radius,
+    border_color,
+  } = options;
+
+  if (blacklist.includes(username)) {
+    return renderError("Something went wrong");
+  }
```

Comment:
Maybe we should change this to a `throw` instead of a `return`. That way, users can choose to show the error card or not. Currently, if I use the NPM package, this is what my code will look like:

```js
import express from "express";
import { generateStatsCard } from "@zo-bro-23/github-readme-stats-test";

const app = express();

app.get("/", (req, res) => {
  res.setHeader("Content-Type", "image/svg+xml");
  generateStatsCard({ username: "Zo-Bro-23" }).then((card) => {
    res.send(card);
  });
});

app.listen(8000);
```

The `.then()` gets fired for errors too, and even if I added a `.catch()`, it would never get fired. Separating errors and successful responses might be the best way to move forward.

Reply:
It makes sense when people use this as an NPM package. 👍🏻 I think @anuraghazra added it this way to fix some renovate problems. 🤔